sql/src/main/java/io/crate/execution/jobs/CountTask.java

```diff
@@ -65,10 +62,13 @@ public synchronized void innerStart() {
             throw new RuntimeException(e);
         }
         countFuture.whenComplete((rowCount, failure) -> {
-            BatchIterator<Row> iterator =
-                rowCount == null ? null : InMemoryBatchIterator.of(new Row1(rowCount), SENTINEL);
-            consumer.accept(iterator, failure);
-            close(failure);
+            if (rowCount == null) {
```

Comment:
having a rowCount != null and a failure cannot happen (like failure happens after rowCount is set)?

Reply:
Shouldn't happen, the semantic is the same as before. But I can also reverse the check to be on the error